### DISCUSSION TITLE
Statically typing schema

### DISCUSSION THREAD

**Discussion by haldunanil** [colinhacks/zod#1410] [Q&A] [answered]
Hi, apologies if this has already been asked/answered or if it's in the readme, but is there a way to assert that a schema adheres to a desired shape rather than inferring the shape from a schema?

For example, I want to do:

```ts
interface MyType {
  id: string;
}

const schema: <something that uses MyType> = z.object({
  id: z.string()
});
// -> passes

const schema: <something that uses MyType> = z.object({
  id: z.string(),
  inventory: z.number(),
});
// -> fails
```

is there something i can put instead of `<something that uses MyType>` that would ensure that the assertions above work as intended?

**Comment by nullndr** [comment, marked as answer]
Yes, check [`toZod`](https://github.com/colinhacks/tozod)
